src/types/field/baseField.ts

```diff
@@ -47,7 +47,7 @@ export interface IFieldSchema extends IField, Document {
 // needs it as a variable to store the client's answer to a field.
 // Hence we need this interface for client-side fields.
 export interface IClientFieldSchema extends IFieldSchema {
-  fieldValue: string
+  fieldValue: string | string[] | boolean[]
```

Comment:
it seems we are trying to generalise this to all field types, but clearly `IClientFieldSchema` is only used for logic. maybe we just need to generalise it to logic fields.